ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageSievingInputStream.java

```diff
@@ -153,80 +193,85 @@ public boolean hasInvalidMessages() {
   }
 
   public List<MessageInfo> getValidMessageInfoList() {
-    return validMessageInfoList;
+    return sievedMessageInfoList;
   }
 
   /**
-   * Ensures blob validity in the given input stream. For now, blobs are checked for message corruption
-   * @param byteArrayInputStream stream against which validation has to be done
-   * @param size total size of the message expected
-   * @param currentOffset Current offset at which the data has to be read from the given byte array
+   * Validates and potentially transforms the given input stream consisting of message data. Blobs are validated for
+   * message corruption and acceptable formats.
+   * @param msg byte array containing the message that needs to be validated and possibly transformed.
+   * @param msgInfo the {@link MessageInfo} associated with the message.
    * @param storeKeyFactory StoreKeyFactory used to get store key
-   * @return true if message is valid and false otherwise
+   * @param replacementKey the converted {@link StoreKey} to replace the one in msgInfo with. This can also be the same
+   *                       key as the original (if no transformation is required).
+   * @return {@link PutMessageFormatInputStream} a validated and possibly transformed message stream, if message is
+   *         valid and successfully transformed; null otherwise
    * @throws IOException
    */
-  private boolean checkForMessageValidity(ByteArrayInputStream byteArrayInputStream, int currentOffset, long size,
-      StoreKeyFactory storeKeyFactory, MessageInfo msgInfo) throws IOException {
-    boolean isValid = false;
+  private PutMessageFormatInputStream validateAndTransform(byte[] msg, MessageInfo msgInfo,
```

Comment:
why not use `List<Transformer>` like the `StoreCopier` here (might need to move that class)? I think that will be the more extensible design (if we servers were to be rewritten)

Reply:
Would simplify testing too because the `Transformer` impls would come tested. This class just has to be tested to make sure it is using the `Transfomer` (s)

Reply:
Made this change, and made a few changes to the `Transformer` interface as well. Let me know what you think.